packages/component-elife-app/server/policies/isAuthor.test.js

```diff
@@ -0,0 +1,58 @@
+const {
+  withAuthsomeMiddleware,
+  AUTHORIZATION_ERROR_MESSAGE,
+} = require('../withAuthsomeMiddleware')
+const { AuthorizationError } = require('@pubsweet/errors')
+
+describe('policy: isAuthor', async () => {
+  const resolver1Name = 'bobby'
+  const mockResolver = jest.fn(() => ({ authorized: true }))
+  const mockManuscript = {
+    findByIdAndAuthor: jest.fn(async (m, u) => {
+      if (u === 'a_valid_user' && m === 'exampleManuscriptId') {
+        return { ok: true }
+      } 
+        throw new Error('some error')
+      
+    }),
+  }
+
+  const mockUseCases = {
+    [`${resolver1Name}UseCase`]: {
+      authsomePolicies: ['isAuthenticated', 'isAuthor'],
+    },
+  }
+
+  const mockResolvers = {
+    Mutation: {
+      [resolver1Name]: mockResolver,
+    },
+  }
+
+  const modifiedResolvers = withAuthsomeMiddleware(
+    mockResolvers,
+    mockUseCases,
+    { _models: { Manuscript: mockManuscript } },
+  )
+
+  it('allows the author of a manuscript to access the resource', async done => {
+    await expect(
+      modifiedResolvers.Mutation[resolver1Name](
+        { original: 'params' },
+        { manuscriptId: 'exampleManuscriptId' },
+        { user: 'a_valid_user' },
+      ),
+    ).resolves.toEqual({ authorized: true })
+    done()
```

Comment:
I'm not sure done is necessary when the test itself is being run as an async with awaits

Reply:
I'm pretty sure it's required when the function passed to `it` is `async`

Reply:
Wait no it's not needed